frontend/javascripts/admin/dataset/dataset_add_remote_view.tsx

```diff
@@ -367,15 +443,19 @@ function AddRemoteLayer({
   uploadableDatastores,
   setDatasourceConfigStr,
   onSuccess,
+  onError,
   dataSourceEditMode,
+  defaultUrl,
 }: {
   form: FormInstance;
   uploadableDatastores: APIDataStore[];
   setDatasourceConfigStr: (dataSourceJson: string) => void;
-  onSuccess?: () => void;
+  onSuccess?: (datasetUrl?: string) => Promise<void> | void;
```

Comment:
in `dataset_add_remote_view` there is a case where `onSuccess={() => setShowAddLayerModal(false)}`

Reply:
ok, I see. however, I wouldn't adapt the interface to that only because one "consumer" does not need the url. otherwise, all other consumers need to handle the case where the url does not exist, even though it always exists.
therefore, my suggestion would be:
- change the interface to `onSuccess?: (datasetUrl: string) => Promise<void> | void;`
- change the spots where the url is not needed to `onSuccess={(_datasetUrl: string) => setShowAddLayerModal(false)}`
- check that the other onSuccess definitions don't have any obsolete null-checks (also in other functions that are potentially called with the url)

if this doesn't make any sense, please let me know. maybe I'm missing something obvious :sweat_smile:

Reply:
no, that sounds good. While changing the code as you suggested, I noticed that the type checker doesnt complain about `onSuccess={() => setShowAddLayerModal(false)}` even after having changed the interface (I double-checked that its actutally changed). Tbh I don't understand why.

Reply:
even better :+1: I assume that this case is tolerated because the called function is not obligated to use all params. the opposite (calling a function with a missing parameter) should be an actual error.